presto-main/src/main/java/com/facebook/presto/operator/scalar/DayOfMonthWithEnglishSuffixPrinter.java

```diff
@@ -0,0 +1,192 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.scalar;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableMap;
+import org.joda.time.Chronology;
+import org.joda.time.DateTimeFieldType;
+import org.joda.time.DateTimeZone;
+import org.joda.time.ReadablePartial;
+import org.joda.time.format.DateTimeParser;
+import org.joda.time.format.DateTimeParserBucket;
+import org.joda.time.format.DateTimePrinter;
+import org.joda.time.format.FormatUtils;
+
+import java.io.IOException;
+import java.io.Writer;
+import java.util.Locale;
+import java.util.Map;
+
+public final class DayOfMonthWithEnglishSuffixPrinter
+        implements DateTimePrinter, DateTimeParser
+{
+    private static final int MIN_PRINTED_LENGTH = 3;
+    private static final int MAX_PRINTED_LENGTH = 4;
+    private static final DateTimeFieldType DAY_OF_MONTH = DateTimeFieldType.dayOfMonth();
+    private static final Map<String, Integer> THREE_LETTER_PARSE_LUT = new ImmutableMap.Builder<String, Integer>()
+            .put("1st", 1)
+            .put("2nd", 2)
+            .put("3rd", 3)
+            .put("4th", 4)
+            .put("5th", 5)
+            .put("6th", 6)
+            .put("7th", 7)
+            .put("8th", 8)
+            .put("9th", 9)
+            .build();
+    private static final Map<String, Integer> FOUR_LETTER_PARSE_LUT = new ImmutableMap.Builder<String, Integer>()
+            .put("10th", 10)
+            .put("11th", 11)
+            .put("12th", 12)
+            .put("13th", 13)
+            .put("14th", 14)
+            .put("15th", 15)
+            .put("16th", 16)
+            .put("17th", 17)
+            .put("18th", 18)
+            .put("19th", 19)
+            .put("20th", 20)
+            .put("21st", 21)
+            .put("22nd", 22)
+            .put("23rd", 23)
+            .put("24th", 24)
+            .put("25th", 25)
+            .put("26th", 26)
+            .put("27th", 27)
+            .put("28th", 28)
+            .put("29th", 29)
+            .put("30th", 30)
+            .put("31st", 31)
+            .build();
+
+    @Override
+    public int estimatePrintedLength()
+    {
+        return MAX_PRINTED_LENGTH;
+    }
+
+    private static String getSuffix(int day)
```

Comment:
We already have a table of values for parsing. Using these for printing is simpler and avoids errors.